iroh-store/src/main.rs

```diff
@@ -16,10 +15,10 @@ async fn main() -> anyhow::Result<()> {
     let version = env!("CARGO_PKG_VERSION");
     println!("Starting iroh-store, version {version}");
 
-    let sources = vec![iroh_home_path(CONFIG_FILE_NAME), args.cfg.clone()];
+    let sources = vec![iroh_config_path(CONFIG_FILE_NAME), args.cfg.clone()];
     let config = make_config(
         // default
-        Config::new_grpc(args.path.clone().unwrap_or_else(|| PathBuf::from(""))),
+        Config::new_grpc(config_data_path(args.path.clone())),
```

Comment:
testing locally on a mac rockDB `create_if_missing` option creates all nested folders, so feeding this `iroh/store` `PathBuf` worked without any additional changes